src/Tribe/Views/V2/Views/Traits/HTML_Cache.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * Provides methods for HTML caching for views
+ *
+ * @since   4.9.11
+ *
+ * @package Tribe\Events\Views\V2\Views
+ */
+
+namespace Tribe\Events\Views\V2\Views\Traits;
+
+use Tribe__Cache_Listener as Cache_Listener;
+use Tribe__Context as Context;
+use Tribe__Date_Utils as Dates;
+
+/**
+ * Trait HTML_Cache
+ *
+ * @since   5.0.0
+ *
+ * @package Tribe\Events\Views\V2\Views
+ *
+ * @property Context $context The current View context.
+ */
+trait HTML_Cache {
+	/**
+	 * Return cached HTML if enabled and cache is set.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return null|string
+	 */
+	public function maybe_get_cached_html() {
+
+		if ( ! $this->should_cache_html() ) {
+			return;
+		}
+
+		$cache_key = $this->get_cache_html_key();
+
+		if ( $cached_html = tribe( 'cache' )->get_transient( $cache_key, $this->cache_html_triggers() ) ) {
+			return $cached_html;
+		}
+	}
+
+	/**
+	 * If caching is enabled, set the cache.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param string $html HTML markup for view.
+	 */
+	public function maybe_cache_html( $html ) {
+		/**
+		 * Filter the cache TTL
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param int     $cache_ttl Cache time to live.
+		 * @param Context $context   The View current context.
+		 * @param View    $this      The current View instance.
+		 */
+		$cache_expiration = apply_filters( 'tribe_events_views_v2_cache_html_expiration', DAY_IN_SECONDS, $this->get_context(), $this );
+
+		$cache_key = $this->get_cache_html_key();
+
+		if ( ! $this->should_cache_html() ) {
+			return;
+		}
+
+		tribe( 'cache' )->set_transient( $cache_key, $html, $cache_expiration, $this->cache_html_triggers() );
+	}
+
+	/**
+	 * Fetch the HTML cache triggers.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return array
+	 */
+	protected function cache_html_triggers() {
+		$triggers = [
+			Cache_Listener::TRIGGER_SAVE_POST,
+			Cache_Listener::TRIGGER_UPDATED_OPTION,
+			Cache_Listener::TRIGGER_GENERATE_REWRITE_RULES,
+		];
+
+		return $triggers;
+	}
+
+	/**
+	 * Determine if HTML of the current view needs to be cached.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return bool
+	 */
+	public function should_cache_html() {
+		$context = $this->get_context();
+
+		$cached_views = [
+			'month',
```

Comment:
:no_entry_sign: Short array syntax is not allowed

<sub>hash: 7d23d575433345ad7c7b8f480a08a93d</sub>